### PR TITLE
Fix collision squared distance computation

### PR DESCRIPTION
**Pull request by CedricGuillemet**
Followup on https://forum.babylonjs.com/t/jumping-camera-movement/18352
Fixes issues introduced with https://github.com/BabylonJS/Babylon.js/pull/9873

Side note: a PR claiming 'optimize something ... ' shall not be merged unless there are metrics clearly showing a performance improvement. I doubt a significant improvement can benefit from that PR. Moreover, adding more states for the same value (here, velocity) almost always make the work more difficult as values must be properly synced.

- `distToCollisionSquared` is not `t * this._velocitySquaredLength` but `(t * velocityLength) * (t * velocityLength)` which is `t2 * _velocitySquaredLength`

### CODE REVIEW

**Comment by Popov72** [comment]
> (t * velocityLength) * (t * velocityLength)

You probably meant: `(t + velocityLength) * (t + velocityLength)`

File: src/Collisions/collider.ts [hunk 1]
```diff
@@ -260,6 +260,7 @@ export class Collider {
         if (!embeddedInPlane) {
             this._basePoint.subtractToRef(trianglePlane.normal, this._planeIntersectionPoint);
             this._velocity.scaleToRef(t0, this._tempVector);
+            this._velocitySquaredLength = this._velocity.lengthSquared();
```
Comment (Popov72) [inline src/Collisions/collider.ts:263]: Is this line necessary? It is already done in `initialize` (I guess this function is always called before `_testTriangle`?) and `_velocity` is not modified anywhere in the class (and is private, so can't be modified by the outside), so the value computed in `initialize` should still be the right one.

**Comment by CedricGuillemet** [comment]
@Popov72 No, actually, my simplification was not correct. It's actualy t ^ 2 * distanceSquared

File: src/Collisions/collider.ts (hunk 1 above)
Comment (CedricGuillemet) [reply src/Collisions/collider.ts]: Good catch! removed now

**Comment by deltakosh** [comment]
can we merge?

**Comment by Popov72** [comment]
That's ok for me.

**Pull request merged**